pilot/pkg/bootstrap/certcontroller.go

```diff
@@ -110,14 +111,29 @@ func (s *Server) initCertController(args *PilotArgs) error {
 //
 // TODO: If the discovery address in mesh.yaml is set to port 15012 (XDS-with-DNS-certs) and the name
 // matches the k8s namespace, failure to start DNS server is a fatal error.
-func (s *Server) initDNSCerts(hostname, namespace string) error {
+func (s *Server) initDNSCerts(hostname, customHostname, namespace string) error {
```

Comment:
In mesh config we have a whole section to configure what SANs should end up in each cert. 
I don't think we need a new API...

Reply:
The API is generating secrets and saving them to certs though. I agree we should probably reuse it but not sure exactly how that should be handled. Probably the old case of saving to secrets should be removed

Reply:
We no longer save secrets. I don't think the API depend on 'saving' - it was an implementation detail. It should still save for ingress if 'SDS ingress' is off - but I think the plans are to also move ingress to SDS-only ( with SDS reading files ).

If the API doc mentions saving - we should fix it.

Reply:
I am pretty sure we are still saving secrets - haven't tried it but I look at the code recently. Is it intended for people to be using this for Ingress certs? I thought this was for webhooks mostly. Seems like something cert manager should do not istio

Reply:
I'm not seeing such an API in mesh config.  please advise with exact API and how to use it if it exists @costinm.

Reply:
Thanks!